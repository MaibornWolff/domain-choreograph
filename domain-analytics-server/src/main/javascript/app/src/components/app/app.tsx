import { Global, css } from '@emotion/core';
import * as React from 'react';
import { Redirect, Route, RouteComponentProps, Switch } from 'react-router';
import { ConnectedRouter } from 'connected-react-router';
import { DropMessage } from '~components/drop-message/drop-message';
import { DropZoneContainer } from '~containers/drop-zone.container';
import { GraphPageContainer } from '~containers/graph-page.container';
import { history } from '~history';
import { styled } from '~utils/styled';

const globalStyles = css`
  html {
    font-family: 'Roboto', sans-serif;
    font-size: 16px;
  }

  html, body {
    padding: 0;
    margin: 0;
  }

  a {
    color: inherit;
    text-decoration: none;
  }
`;

const Wrapper = styled('div')`
  color: ${({ theme }) => theme.colors.text.normal.primary};
  background: ${({ theme }) => theme.colors.background[0]};
  overflow: hidden;
`;

const ContentWrapper = styled('div')`
  position: relative;
  height: 100vh;
  width: 100vw;
  display: flex;
`;

export interface AppProps {
  loadedGraph: boolean;
  resetState: () => void;
  fetchGraph: () => void;
}

export interface AppState {
}

export class App extends React.Component<AppProps, AppState> {
  public render() {
    return (
      <Wrapper>
        <Global styles={globalStyles}/>
        <DropZoneContainer>
          <ConnectedRouter history={history}>
            <Switch>
              <Route path="/load" component={() =>
                this.props.loadedGraph
                  ? <Redirect to={'/Application'}/>
                  : <DropMessage>Drop json here</DropMessage>}
              />
              {this.props.loadedGraph ? null : <Redirect to={'/load'}/>}
              <Route path="/:executionContext/:scope" component={renderGraphPage}/>
              <Route path="/:executionContext" component={renderGraphPage}/>
              <Route path="/" component={() => <Redirect to="/Application"/>}/>
            </Switch>
          </ConnectedRouter>
        </DropZoneContainer>
      </Wrapper>
    );
  }

  public componentDidCatch(err: Error) {
    this.props.resetState();
    return <Redirect to={'/'}/>;
  }

  public componentDidMount() {
    this.props.fetchGraph();
  }
}

const renderGraphPage: React.StatelessComponent<RouteComponentProps<any>> = ({ match: { params: { executionContext, scope } } }) => {
  return (
    <ContentWrapper>
      <GraphPageContainer executionContext={executionContext} scope={scope}/>
    </ContentWrapper>
  );
};
