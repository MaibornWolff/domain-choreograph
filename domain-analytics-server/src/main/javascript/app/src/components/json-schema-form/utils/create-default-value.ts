import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';

const defaultValues: Record<Exclude<JSONSchema4TypeName, 'object'>, any> = {
  string: undefined,
  number: undefined,
  integer: undefined,
  boolean: undefined,
  array: [],
  'null': null,
  any: ''
};

export function createDefaultValue(schema: JSONSchema4): any {
  const type = schema.type || 'any';
  if (type instanceof Array) {
    throw new Error('Array types are not supported');
  } else if (type === 'object') {
    return createDefaultValueForObject(schema);
  } else {
    return defaultValues[type];
  }
}

function createDefaultValueForObject(schema: JSONSchema4): Promise<object> {
  if (schema.type !== 'object') {
    throw new Error('Expect schema to be an object');
  }
  let defaultValue: any = {};
  if (schema.properties) {
    Object.entries(schema.properties).forEach(([key, property]) => {
      defaultValue[key] = createDefaultValue(property)
    });
  }
  return defaultValue;
}

