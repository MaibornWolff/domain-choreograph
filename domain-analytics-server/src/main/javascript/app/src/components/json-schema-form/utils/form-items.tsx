import { styled } from '~utils/styled';

export const FormItems = styled('div')`
    display: flex;
    flex-direction: column;

    & > * {
      margin-bottom: .5rem;
    }
`;
