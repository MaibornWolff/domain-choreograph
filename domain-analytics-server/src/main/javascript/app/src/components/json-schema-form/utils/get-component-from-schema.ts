import { FieldComponent } from 'clean-forms';
import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';
import { DatePickerMsField } from '~components/json-schema-form/form-fields/date-picker-ms-field';
import { NumberTextField, TextField } from '~components/json-schema-form/form-fields/text-field';
import { JsonSchemaFormFieldProps } from '~components/json-schema-form/models/json-schema-form-field-props';

export interface JsonSchemaComponentMap {
  matcher?: { type: JSONSchema4TypeName, format?: string };
  field: FieldComponent<any, JsonSchemaFormFieldProps>;
}

const defaultMapping: JsonSchemaComponentMap[] = [
  { matcher: { type: 'integer', format: 'utc-millisec' }, field: DatePickerMsField },
  { matcher: { type: 'integer' }, field: NumberTextField },
  { matcher: { type: 'number' }, field: NumberTextField },
  { field: TextField }
];

export function getComponentFromSchema(
  schema: JSONSchema4,
  mapping: JsonSchemaComponentMap[] = defaultMapping
): FieldComponent<any, JsonSchemaFormFieldProps> {
  for (const map of mapping) {
    if (map.matcher == null) {
      return map.field;
    }
    const typeIsMatching = map.matcher.type === schema.type;
    const formatIsMatching = map.matcher.format == null || map.matcher.format === schema.format;
    if (typeIsMatching && formatIsMatching) {
      return map.field;
    }
  }
  throw new Error('Couldn\'t match the schema');
}
