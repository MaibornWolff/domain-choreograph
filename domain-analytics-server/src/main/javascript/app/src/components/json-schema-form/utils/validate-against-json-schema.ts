import { ValidationFunction } from 'clean-forms';
import { JSONSchema4 } from 'json-schema';
import { Validator } from 'jsonschema';


export function validateAgainstJsonSchema(schema: JSONSchema4): ValidationFunction {
  const validator = new Validator();
  return ({ value }) => {
    const result = validator.validate(value, schema as any);
    const errors = result.errors.map(error => {
      let message = error.message;
      let property = error.property
        .replace(/^instance\.?/, '');
      let schema = error.schema as JSONSchema4;
      if (error.name === 'required') {
        message = `is required`;
        schema = schema.properties![error.argument];
        property = joinProperties(property, error.argument);
      }

      const title = schema.title || '';
      message = `${title} ${message}`;

      return [ property, message ] as [string, string];
    });
    return errors;
  }
}

function joinProperties(path1: string, path2: string): string {
  if (path1 === '') {
    return path2;
  }
  if (path2 === '') {
    return path1;
  }
  return `${path1}.${path2}`;
}
