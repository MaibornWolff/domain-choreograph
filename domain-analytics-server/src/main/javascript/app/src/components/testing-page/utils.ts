export function escapeJavaClass(javaClass: string) {
  return javaClass.replace(/\./g, '-');
}

export function reverseJavaClassEscape(javaClass: string) {
  return javaClass.replace(/-/g, '.');
}
