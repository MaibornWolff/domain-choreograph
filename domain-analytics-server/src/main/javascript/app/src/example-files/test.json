{
  "nodes": {
    "A": {
      "id": "A",
      "name": "A",
      "scope": "root"
    }
  },
  "dependencies": [],
  "scopes": {
    "root": {
      "id": "root",
      "executionContext": "Application",
      "nodes": [
        "A"
      ]
    }
  },
  "executionContexts": {
    "Application": {
      "id": "Application",
      "scopes": [
        "root"
      ]
    }
  }
}
