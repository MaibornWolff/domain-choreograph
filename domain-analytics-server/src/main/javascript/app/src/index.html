<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Domainchoreograph</title>
</head>
<body>
<div id="app"></div>
</body>
</html>