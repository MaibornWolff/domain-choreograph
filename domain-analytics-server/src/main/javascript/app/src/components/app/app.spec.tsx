import { shallow } from 'enzyme';
import * as React from 'react';
import { Redirect } from 'react-router';
import { App } from '~components/app/app';
import { renderWithProviders } from '~utils/test.utils';

describe('App', () => {
  it('should render if graph is loaded', () => {
    expect(renderWithProviders(<App loadedGraph={true} resetState={() => {
    }} fetchGraph={() => {
    }}/>)).toMatchSnapshot();
  });

  it('should render if graph is not loaded', () => {
    expect(renderWithProviders(<App loadedGraph={true} resetState={() => {
    }} fetchGraph={() => {
    }}/>)).toMatchSnapshot();
  });

  it('should call fetch graph on mount', () => {
    const fetchGraph = jest.fn();

    expect(shallow(<App
      loadedGraph={true}
      resetState={() => {
      }}
      fetchGraph={fetchGraph}/>)
    ).toMatchSnapshot();

    expect(fetchGraph).toHaveBeenCalled();
  });

  it('should catch errors', () => {
    const resetState = jest.fn();
    const app = shallow(<App loadedGraph={false} resetState={resetState} fetchGraph={() => {}}/>);

    const errorFallback = app.instance().componentDidCatch!(new Error('test'), {} as any);

    expect(resetState).toHaveBeenCalled();
    expect(errorFallback).toEqual(<Redirect to="/"/>);
  });
});
