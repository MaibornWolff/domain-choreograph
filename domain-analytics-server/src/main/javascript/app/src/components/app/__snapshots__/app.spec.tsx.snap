// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should call fetch graph on mount 1`] = `
<styled.div>
  <GlobalStyleComponent />
  <React.Memo>
    <Connect(ConnectedRouterWithContext)
      createHistory={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Switch>
        <Route
          component={[Function]}
          path="/testing"
        />
        <Route
          component={[Function]}
          path="/load"
        />
        <Route
          component={[Function]}
          path="/:executionContext/:scope"
        />
        <Route
          component={[Function]}
          path="/:executionContext"
        />
        <Route
          component={[Function]}
          path="/"
        />
      </Switch>
    </Connect(ConnectedRouterWithContext)>
  </React.Memo>
</styled.div>
`;

exports[`App should render if graph is loaded 1`] = `
.c1 {
  height: 100vh;
  width: 100vw;
}

.c0 {
  color: rgba(0,0,0,.87);
  background: #E0E0E0;
  overflow: hidden;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;

exports[`App should render if graph is not loaded 1`] = `
.c1 {
  height: 100vh;
  width: 100vw;
}

.c0 {
  color: rgba(0,0,0,.87);
  background: #E0E0E0;
  overflow: hidden;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;
